Src/Couchbase.Linq/BucketContext.cs

```diff
@@ -32,13 +30,16 @@ public class BucketContext : IBucketContext, IChangeTrackableContext
         /// </summary>
         public bool ChangeTrackingEnabled { get { return _beginChangeTrackingCount > 0; } }
 
+        public IDataStore DataStore { get; private set; }
+
         /// <summary>
         /// Creates a new BucketContext for a given Couchbase bucket.
         /// </summary>
         /// <param name="bucket">Bucket referenced by the new BucketContext.</param>
         public BucketContext(IBucket bucket)
         {
             _bucket = bucket;
+            DataStore = new BucketDataStore(bucket);
```

Comment:
Not to be a PITA, but upon seeing this I am not sure if we need a separate IDataStore abstraction; `Execute(...)` might be better off just being a member of IBucketContext. In general wrapping the IBucket in the ctor is going to be a bad decision later on. Perhaps a separate ctor might make it better: 

`public BucketContext(IDataStore dataStore)` 

and 

`public `BucketContext(IBucket bucket) : this(new BucketDataStore(bucket)`

But, every where else in `BucketContext` we use the `IBucket` implementation itself...it seems like duplication of responsibility and may make mocking/testing more complex (one more interface/layer to mock). Additionally, we're losing the ability to manipulate `QueryRequest` parameters as well; we should have an overload of `Execute` to handle  that as well or provide (or design a path towards) an equivalent  so that these values can be set. A good example of a value people will want to set is QueryRequest.Timeout.

-Jeff